File: programming-concepts/selection/python/show_error_if.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4

def show_error():
    """Example of an if/elif statement to match the error code"""
    error_code = input("Enter the error code: ")
    
    if error_code == "400":
        print("Bad request")
    elif error_code == "401":
        print("Unauthorised request")
    elif error_code == "403":
        print("Forbidden request")
    elif error_code == "404":
        print("Page not found")
    elif error_code == "408":
        print("Timeout error")
    else:
        print("Unknown error")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    show_error()

File: programming-concepts/selection/c-sharp/show_error_if.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{

    class Selection
    {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            ShowError();
        }


        // Example of an if/else if statement to match the error code
        public static void ShowError() {
            Console.WriteLine("Enter the error code: ");
            string errorCode = Console.ReadLine();

            if (errorCode == "400") {
                Console.WriteLine("Bad request");
            }
            else if (errorCode == "401") {
                Console.WriteLine("Unauthorised request");
            }
            else if (errorCode == "403") {
                Console.WriteLine("Forbidden request");
            }
            else if (errorCode == "404") {
                Console.WriteLine("Page not found");
            }
            else if (errorCode == "408") {
                Console.WriteLine("Timeout error");
            }
            else {
                Console.WriteLine("Unknown error");
            }
        }
        

    }
}
